dev/benchmarks/complex_layout/lib/main.dart

```diff
@@ -36,13 +37,45 @@ class ComplexLayoutAppState extends State<ComplexLayoutApp> {
     });
   }
 
+  ScrollMode _scrollMode = ScrollMode.complex;
+  ScrollMode get scrollMode => _scrollMode;
+  set scrollMode(ScrollMode mode) {
+    setState(() {
+      _scrollMode = mode;
+    });
+  }
+
   void toggleAnimationSpeed() {
     setState(() {
       timeDilation = (timeDilation != 1.0) ? 1.0 : 5.0;
     });
   }
 }
 
+class TileScrollLayout extends StatelessWidget {
+  const TileScrollLayout({ Key key }) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    final List<Widget> children = <Widget>[];
+    for (int i = 0; i < 200; i++) {
```

Comment:
Lists that have 200+ children will typically not have any of this defined in compile time; it will be 200 model objects (emails/users/whatever) that need to be represented in a list. It is quite intuitive to call setState() on a stateful widget when data is available which makes build() function an ideal place to build widget tree from provided data. My understanding of "viewporting" in the new List implementation is that build() _doesn't_ get called in every frame. Instead, ListView is provided with a full list of widgets upfront and it handles scrolling internally attaching/detaching these widget trees dynamically.

It seems that this understanding is wrong and we have to write a delegate that returns how many objects are present in the list and can return list items dynamically given indices (similar to RecyclerView.Adapter). Please confirm.